lib/linter/linter.js

```diff
@@ -1172,9 +1172,9 @@ class Linter {
      * Initialize the Linter.
      * @param {Object} [config] the config object
      * @param {string} [config.cwd] path to a directory that should be considered as the current working directory, can be undefined.
-     * @param {"flat"|"eslintrc"} [config.configType="eslintrc"] the type of config used.
+     * @param {"flat"|"eslintrc"} [config.configType="flat"] the type of config used.
      */
-    constructor({ cwd, configType } = {}) {
+    constructor({ cwd, configType = "flat" } = {}) {
```

Comment:
When `config` argument is not passed to `verify`/`verifyAndFix`, Linter still runs eslintrc code paths, which is probably not what we want if `"flat"` is the new default.

An observable difference:

```js
const { Linter } = require("eslint");

const linter = new Linter();

const results = linter.verify(`

    /* eslint no-undef: 2 */
    /* eslint-env browser */

    window;

`);

console.log(results); // [], but there should be a no-undef lint message in flat config mode

```

The cause is the condition it uses to switch between modes:

https://github.com/eslint/eslint/blob/a9a17b3f1cb6b6c609bda86a618ac5ff631285d2/lib/linter/linter.js#L1371-L1372

Reply:
Hmmm, in that case adding a default value to the constructor won't fix the problem, because we are still checking for `config`. Probably need to revisit all of that logic.